src/ngraph/pass/serialize.cpp

```diff
@@ -0,0 +1,45 @@
+/*******************************************************************************
+* Copyright 2017-2018 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include <fstream>
+
+#include "ngraph/file_util.hpp"
+#include "ngraph/pass/serialize.hpp"
+#include "ngraph/serializer.hpp"
+#include "ngraph/util.hpp"
+#include "nlohmann/json.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+pass::Serialization::Serialization(const string& output_file)
+    : m_prefix_filename{output_file}
+{
+}
+
+bool pass::Serialization::run_on_module(vector<shared_ptr<Function>>& functions)
+{
+    for (shared_ptr<Function> f : functions)
+    {
+        string js = serialize(f, 4);
```

Comment:
There is a serialize call that can serialize directly to a file.
`void serialize(const std::string& path,
                   std::shared_ptr<ngraph::Function> func,
                   size_t indent = 0);`